src/test/java/com/googlecode/jsonrpc4j/server/JsonRpcServerTest.java

```diff
@@ -252,5 +252,31 @@ public void testGzipRequestAndResponse() throws IOException {
 	public interface ServiceInterface {
 		String testMethod(String param1);
 	}
-	
+
+	@Test
+	public void testGetMethod_longId() throws Exception {
```

Comment:
I recommend moving this test into `JsonRpcBasicServerTest`, which is where the rest of the ID parsing logic is tested. In there you can create a test like this:

```java
@Test
public void idLongType() throws Exception {
    EasyMock.expect(mockService.testMethod(param1)).andReturn(param1);
    EasyMock.replay(mockService);
    jsonRpcServer.handleRequest(messageWithListParamsStream(1600000000000L, "testMethod", param1), byteArrayOutputStream);
    assertTrue(decodeAnswer(byteArrayOutputStream).get(ID).isLong());
}
```

which avoids having to parse the JSON via string manipulation.

Reply:
The test that you suggest is exactly what I tried, but it fails for a different reason. Do you think you can take a look?

https://github.com/briandilley/jsonrpc4j/pull/185/files#diff-af2392062980e71e9fbd324d1a06f191R190